python/TestHarness/validation/ValidationCase.py

```diff
@@ -194,57 +258,257 @@ def addResult(self, status: Status, message: str, **kwargs) -> None:
         Keyword arguments:
-            Additional arguments passed to FloatData
+            Additional arguments passed to ScalarData (bounds, nominal, etc)
         """
-        if not isinstance(value, float):
-            raise ValueError('value is not of type float')
+        value = self.toFloat(value, 'value')
+        if not isinstance(description, str):
+            raise TypeError('description: not of type str')
+        if units is not None and not isinstance(units, str):
+            raise TypeError('units: not of type str or None')
+        bounds = kwargs.get('bounds')
+        if bounds is not None:
+            if not isinstance(bounds, tuple):
+                raise TypeError('bounds: not of type tuple')
+            if not len(bounds) == 2:
+                raise TypeError('bounds: not of length 2 (min and max)')
+            kwargs['bounds'] = (self.toFloat(bounds[0], 'bounds min'),
+                                self.toFloat(bounds[1], 'bounds max'))
+        nominal = kwargs.get('nominal')
+        if nominal is not None:
+            kwargs['nominal'] = self.toFloat(nominal, 'nominal')
+
+        data = self._addData(self.ScalarData, key, value, description, units=units, **kwargs)
 
-        if key in self._data:
-            raise DataKeyAlreadyExists(key)
+        result_kwargs = {'data_key': key,
+                         'validation': kwargs.pop('validation', True)}
 
-        data = self.FloatData(value=value,
-                              units=units,
-                              key=key,
-                              description=description,
-                              test=self._current_test,
-                              **kwargs)
-        self._data[key] = data
+        if data.bounds is not None:
+            status, message = self.checkBounds(data.value, data.bounds[0],
+                                               data.bounds[1], data.units)
+            self.addResult(status, message, **result_kwargs)
+
+    @staticmethod
+    def toListFloat(value: typing.Any, context: Optional[str] = None) -> list[float]:
+        """
+        Attempts to convert the given value to a one-dimensional
+        list of floating point values
+
+        Args:
+            value: The number-like list value to convert
+            context (optional): Optional context to prefix exceptions with
+        Returns:
+            list[float]: The converted value
+        """
+        prefix = f'{context} ' if context else ''
+        try:
+            arr = np.array(value, dtype=np.float64)
+        except Exception as e:
+            raise TypeError(f'{prefix}array conversion failed') from e
+        if arr.ndim != 1:
+            raise TypeError(f'{prefix}not one-dimensional')
+        value = arr.tolist()
+        for i, v in enumerate(value):
+            if math.isnan(v):
+                raise ValueError(f'{prefix}value at index {i} is nan')
+        return value
```

Comment:
```suggestion
        nan_ind = np.argwhere(np.isnan(arr)).reshape((-1))
        if nan_ind.any():
            raise ValueError(f'{prefix}values at indices {nan_ind} are nan')
        return arr.tolist()
```

Reply:
By `if nan_ind.any()`, you mean `nan_ind.size > 0` :)